assets/js/components/timer.js

```diff
@@ -0,0 +1,39 @@
+import React from 'react'
+import moment from 'moment'
+
+class Timer extends React.Component {
+  state = { date: Date.now() }
+
+  componentWillMount () {
+    this.interval = window.setInterval(() => {
+      this.setState({ date: Date.now() })
+    }, 1000)
+  }
+
+  componentWillUnmount () {
+    window.clearInterval(this.interval)
+  }
+
+  showDate (difference, d, h, m, s) {
+    if (difference > 0) {
+      return `${d}d ${h}h ${m}m ${s}s left untill next TOURNAMENT`
+    } else {
+      return 'EXPIRED'
+    }
+  }
+
+  render () {
+    const date = new Date(this.state.date)
+    const startDate = new Date(this.props.startDate)
+    const difference = Math.floor(startDate.getTime() - date.getTime())
+    let {days, hours, minutes, seconds} = moment.duration(difference)._data
```

Comment:
You can actually let moment calculate the difference and the format it as you wish:
- diffing https://stackoverflow.com/a/25150793/5670873,
- formatting https://momentjs.com/docs/#/displaying/format/.

Reply:
Disclaimer: I didn't know that before. 🙈